lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -814,6 +814,16 @@ SPIRVType *LLVMToSPIRVBase::transSPIRVOpaqueType(StringRef STName,
 SPIRVType *LLVMToSPIRVBase::transScavengedType(Value *V) {
   if (auto *F = dyn_cast<Function>(V)) {
     FunctionType *FnTy = Scavenger->getFunctionType(F);
+    // VarArg functions other than printf are not supported in SPIR-V.
+    if (FnTy->isVarArg()) {
+      StringRef DemangledName;
+      oclIsBuiltin(F->getName(), DemangledName);
+      BM->getErrorLog().checkError(
+          (DemangledName.starts_with("printf(") ||
+           DemangledName.starts_with("__spirv_ocl_printf(")),
```

Comment:
oclIsBuiltin demangles "printf" to "__spirv_ocl_printf".  I think only checking for:  DemangledName.starts_with("__spirv_ocl_printf") should be sufficient.  I think we should possibly be able to check for equality instead of starts_with.

Also does DemangledName include an open parenthesis?

Reply:
I have actually realized none of `printf` variants can reach this piece of code because they are skipped in https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/7dacb7cdedc9c01e0ce88f1f3766c42ab8f59b7c/lib/SPIRV/SPIRVWriter.cpp#L5789. More specifically, `isBuiltinTransToExtInst(&F)` returns true for `printf`, so it is just skipped and never added to neither `Decls` nor `Defs`. Considering this, I think we should be safe by just checking if the function is variadic to report the error at this point. What do you think? Also tagging the rest of approvers to gather their opinions: @svenvh @MrSidims @VyacheslavLevytskyy